units/en/unitbonus3/rl-for-trading.mdx

```diff
@@ -0,0 +1,162 @@
+
+<figure>
+<img src="https://github.com/user-attachments/assets/60e0173b-c88a-4981-8da8-fb81e9f0e0b4" alt="FinRL Framework"/>
+<figcaption>Source: [FinRL GitHub repository](https://github.com/AI4Finance-Foundation/FinRL)</figcaption>
+</figure>
+
+1. **Market Environments: **FinRL provides gym-style market environments that simulate various stock markets, such as NASDAQ-100, DJIA, S&P 500, HSI, SSE 50, and CSI 300. These environments are configured with historical stock market datasets.
+2. **Agents: **The framework includes state-of-the-art deep reinforcement learning (DRL) algorithms like DQN, DDPG, PPO, SAC, A2C, and TD3. These agents are trained using neural networks to make trading decisions.
+3. **Applications: **FinRL supports various trading tasks, including single stock trading, multiple stock trading, and portfolio allocation. It also incorporates important trading constraints such as transaction costs, market liquidity, and risk aversion.
+
+### Example Workflow
+
+1. **Data Preparation:** Download historical stock data using libraries like Yahoo Finance API.
+
+```python
+from finrl.meta.preprocessor.yahoodownloader import YahooDownloader
+df = YahooDownloader(start_date='2009-01-01', end_date='2020-07-01', ticker_list=['AAPL', 'AMZN']).fetch_data()
+
+# Split df into df_train and df_test 
+```
+
+2. **Environment Configuration: **Create a trading environment using FinRL's StockTradingEnv class.
+
+```python
+from finrl.config import INDICATORS
+from finrl.meta.env_stock_trading.env_stocktrading import StockTradingEnv
+
+# This is only for illustration.
+# Values such as state_space, stock_dimension should be derived in the notebook before using them here.
+env_kwargs = {
+    "hmax": 100,
+    "initial_amount": 1000000,
+    "num_stock_shares": num_stock_shares,
+    "state_space": state_space,
+    "stock_dim": stock_dimension,
+    "tech_indicator_list": INDICATORS,
+    "action_space": stock_dimension,
+    "reward_scaling": 1e-4
+}
+
+env_train = StockTradingEnv(df=df_train, **env_kwargs)
+```
+
+3. **Agent Training: **Train a DRL agent using algorithms like DDPG or PPO from stable-baselines.
+
+There is also an option to train agent using ensemble strategy.
+
+```python
+from finrl.agents.stablebaselines3.models import DRLAgent
+agent = DRLAgent(env=env_train)
+agent.get_model('ppo')
+trained_ppo = agent.train_model()
+trained_ppo.save("trained_models/agent_ppo")
+```
+
+4. **Backtesting and Evaluation: **Evaluate the trained agent's performance using backtesting.
+
+Create a new environment using df_test to simulate test scenario.
+
+```python
+trained_ppo = PPO.load("trained_models/agent_ppo.zip")
```

Comment:
`PPO` is not imported anywhere in the notebook. I guess it can be retrieved e.g. through `finrl.stablebaselines3.models.DRLAgent.get_model("ppo")`.

Reply:
It is imported through stablebaseline3, have included a import statement.